app/src/org/commcare/views/QuestionsView.java

```diff
@@ -320,10 +322,42 @@ private void addHintText(String hintText) {
         }
     }
 
-    public void setFocus(Context context) {
-        if (widgets.size() > 0) {
-            widgets.get(0).setFocus(context);
+    public void setFocus(Context context, int indexOfLastChangedWidget) {
+        QuestionWidget widgetToFocus = null;
+        if (indexOfLastChangedWidget != -1) {
+            widgetToFocus = widgets.get(indexOfLastChangedWidget);
+        } else if (widgets.size() > 0) {
+            widgetToFocus = widgets.get(0);
         }
+        if (widgetToFocus != null) {
```

Comment:
I call both of the methods below in order to focus on any widget because: 
-Scrolling to a widget is needed for any widget that does not involve some sort of text entry (and therefore will not get scrolled to automatically by calling 'setFocus' on it)
-But, `setFocus` still has some smaller side effects for those widgets, such as hiding the keyboard.